e2eAIOK/common/trainer/utils/utils.py

```diff
@@ -189,23 +189,24 @@ def __init__(self, tolerance_epoch = 5, delta=0, is_max = False, metric_threshol
         self.early_stop = False
 
     def __call__(self, validation_metric, optimal_metric):
-        ############## absolute level #################
         if self._metric_threshold is not None:
+            ############## absolute level #################
             if (self._is_max and validation_metric >= self._metric_threshold) or\
             ((not self._is_max) and validation_metric <= self._metric_threshold):
                 self.early_stop = True
                 print("Earlystop when meet metric_threshold [%s]"%self._metric_threshold)
                 logging.info("Earlystop when meet metric_threshold [%s]"%self._metric_threshold)
                 return
-        ############## relative level #################
-        if (self._is_max and (validation_metric < optimal_metric - self._delta)) \
-                or ((not self._is_max) and (validation_metric > optimal_metric + self._delta)): # less optimal
-            self._counter += 1
-        else: # more optimal
-            logging.info("Reset earlystop counter")
-            self._counter = 0
-        if self._counter >= self._tolerance_epoch:
-            self.early_stop = True
+        else:
```

Comment:
@XinyaoWa , when 'else' path will be called? This fix is for which bullet in your description?

Reply:
If not set metric_threshold, else will be called, in this case, training will stop when acc not increase for a while. 

This fixs "For Earlystop in utils.py, disable tolerance when set target metric". Before fix, even metric_threshold is set, "stop when acc not increase for a while" will also be applied, which is harmful when we want to meet some target score; thus fix it.

Reply:
I see, thanks for the explanation.